### DISCUSSION TITLE
Which folders are exposed to the client-side bundle?

### DISCUSSION THREAD

**Discussion by albertbronsky** [vercel/next.js#32984] [Help]
Just wanted to clarify which folders in Next.js are exposed to the client-side bundle?

Documentation says that `/pages/api` (API routes) is protected and is a server-side only bundle. What about the other folders (`lib`, `components`, etc.)?

What about custom folders? 
If I create a folder called `schema` in the root directory to store my `prisma.schema` file in it (and import it in my API routes to make calls to database with Prisma) - will the `schema` folder get leaked to the client-side or not?

**Comment by briowill** [comment]
it depends, the `schema` or any `custom folder` that is excluded from the `pages`/`app` directory won't be directly included in the client side bundle (exposed to the public). 
If the only import is in an `api route ` the schema content won't be exposed, however if a component in a page directory (example pages/home) imports the schema file then the content will be exposed.
